Geometry/GEMGeometryBuilder/src/GEMGeometryBuilderFromDDD.cc

```diff
@@ -122,42 +133,60 @@ void GEMGeometryBuilderFromDDD::build(GEMGeometry& theGeometry,
 
     doSuper = fv.nextSibling();
 
-    if (!loopExecuted)
+    if (!loopExecuted) {
       delete gemChamber;
+    }
   }
 
   // construct the regions, stations and rings.
   for (int re = -1; re <= 1; re = re + 2) {
     GEMRegion* region = new GEMRegion(re);
-    for (int st = 1; st <= GEMDetId::maxStationId; ++st) {
+    for (int st = 0; st <= GEMDetId::maxStationId; ++st) {
       GEMStation* station = new GEMStation(re, st);
       std::string sign(re == -1 ? "-" : "");
-      std::string name("GE" + sign + std::to_string(st) + "/1");
+      std::string name = (st == 0) ? "GE0" : ("GE" + sign + std::to_string(st) + "/1");
       station->setName(name);
+      bool foundSuperChamber = false;
       for (int ri = 1; ri <= 1; ++ri) {
         GEMRing* ring = new GEMRing(re, st, ri);
         for (auto superChamber : superChambers) {
           const GEMDetId detId(superChamber->id());
           if (detId.region() != re || detId.station() != st || detId.ring() != ri)
             continue;
 
-          superChamber->add(
-              theGeometry.chamber(GEMDetId(detId.region(), detId.ring(), detId.station(), 1, detId.chamber(), 0)));
-          superChamber->add(
-              theGeometry.chamber(GEMDetId(detId.region(), detId.ring(), detId.station(), 2, detId.chamber(), 0)));
+          foundSuperChamber = true;
+          int nlayers = (st == 0) ? GEMDetId::maxLayerId0 : GEMDetId::maxLayerId;
+
+          for (int la = 1; la <= nlayers; ++la) {
+            GEMDetId chId(detId.region(), detId.ring(), detId.station(), la, detId.chamber(), 0);
+            auto chamber = theGeometry.chamber(chId);
+            if (!chamber) {
+              edm::LogWarning("GEMGeometryBuilderFromDDD") << "Missing chamber " << chId << std::endl;
+            }
+            superChamber->add(chamber);
+          }
           ring->add(superChamber);
           theGeometry.add(superChamber);
           LogDebug("GEMGeometryBuilderFromDDD") << "Adding super chamber " << detId << " to ring: "
                                                 << "re " << re << " st " << st << " ri " << ri << std::endl;
         }
         LogDebug("GEMGeometryBuilderFromDDD") << "Adding ring " << ri << " to station "
                                               << "re " << re << " st " << st << std::endl;
-        station->add(ring);
-        theGeometry.add(ring);
+        if (!foundSuperChamber) {
+          delete ring;
+        } else {
+          station->add(ring);
+          theGeometry.add(ring);
+        }
+      }
+      if (!foundSuperChamber) {
+        LogDebug("GEMGeometryBuilderFromDDD") << "No superchamber found: re:" << re << " st:" << st << std::endl;
+        delete station;
```

Comment:
After looking through the code more carefully, I see that this PR is fixing a problem with the building of the GEM geometry. Prior to this PR, station 2 was built, even though it was empty. Lines 182-184 ensure the unneeded station is not built.
FYI: @watson-ij @ianna